### PR TITLE
ENH: add methods for domain conversion

### LINKED ISSUE #224: Move python function for based domain conversion from `fmu.sim2seis` to this repo
Move function for depth conversion (and vice versa) from sim2seis repo to here. See also https://github.com/equinor/fmu-sim2seis/issues/30

### PR DESCRIPTION
**Pull request by jcrivenaes**
This was originally in the sim2seis repo, but is migrated to fmu.tools since this function can be applied outside the sim2seis work flow.

The second commit was triggered from that the docs workflow failed, assume to be a consequence of newer Sphinx version.

Closes #224

### CODE REVIEW

**Review by hflesche — COMMENTED** [review]
OK to move from sim2seis to fmu/tools.

File: docs/domain_conversion.rst [hunk 1]
```diff
@@ -0,0 +1,219 @@
+Domain conversion
+=================
+
+In RMS it is possible to do domain conversion using the menu option, but in
+some cases it convinient to be able to convert domains using a script. This
```
Comment (mferrera) [inline docs/domain_conversion.rst:5]:
```suggestion
some cases it convenient to be able to convert domains using a script. This
```
typo

File: src/fmu/tools/domainconversion/dconvert.py [hunk 2]
```diff
@@ -0,0 +1,596 @@
+from __future__ import annotations
```
Comment (mferrera) [inline src/fmu/tools/domainconversion/dconvert.py:1]: I mostly skipped reviewing this file assuming that's well and done from sim2seis

**Review by mferrera — APPROVED** [review]
👍

**Pull request merged**